src/server/req_holdjob.c

```diff
@@ -424,7 +421,7 @@ post_hold(struct work_task *pwt)
 	} else if (code == 0) {
 
 		/* record that MOM has a checkpoint file */
-
+		pjob->ji_qs.ji_substate = JOB_SUBSTATE_RERUN;
```

Comment:
Won't this still have a race condition?  Can the following situation happen:
1) hold is sent to mom
2) job ends and obit is sent
3) job is moved into state 'E'
4) mom replies back to hold request moving the job into substate rerun.

Reply:
@bhroam In this case if mom has sent an obit then it should just reject the hold job sent by server. That is why we now set the substate in rerun when we get a successful reply code from mom.